pkg/endpoint/endpoint.go

```diff
@@ -128,6 +128,10 @@ type Endpoint struct {
 	// recalculated on endpoint restore.
 	createdAt time.Time
 
+	// lastUpdatedTime store the time the endpoint was last changed.
+	// it is used for calculate endpoint progation delay.
```

Comment:
Please update the comment with what is key and what is value here as Joe suggested.

Reply:
hi, @Weil0ng , i have explained why keep this condition, because also need support measure propagation delay of endpoint create when CES disabled, am i wrong ?

Reply:
> @Weil0ng , when CES disabled, CEP create event also call this function for calculate this metirc.
> 
> https://github.com/cilium/cilium/blob/5c23ed43c58353bf9942bc10e8a858b84e913097/pkg/k8s/watchers/cilium_endpoint.go#L44-L50
> 
> , very strange, my commit now has remove the comment, but looks like you can show it

Reply:
By having this condition, you are *only* capturing endpoint creation, the metrics instead is meant for any CEP roundtrip communication which includes update too.

I would suggest removing this so update can also be capture and if you are particularly interested in creations only, you could add a field to the metric to group different types of CEP events.

Reply:
Also you are replying to a comment but you are actually referring to another comment. This creates a lot of confusion for reviewers and other people to look at these later. Please reply to the exact comment you are talking about :)

Reply:
Same here, please leave a comment or address the code change request before you resolve the conversation :)

Reply:
https://github.com/cilium/cilium/blob/85fe8be6ff10f7676a87bae0be77d2ce04f1bd1d/pkg/endpoint/endpoint.go#L131-L134
@Weil0ng ,  you coud give me suggestion if the comment isn't good:)

Reply:
> By having this condition, you are _only_ capturing endpoint creation, the metrics instead is meant for any CEP roundtrip communication which includes update too.
> 
> I would suggest removing this so update can also be capture and if you are particularly interested in creations only, you could add a field to the metric to group different types of CEP events.

in fact,  UpdateFunc method also call endpointUpdated, why just only capturing endpoint creation:)?

Reply:
> @Weil0ng , you coud give me suggestion if the comment isn't good:)

This is from a previous comment from Joe:
Can you also include what the key and value of the map represent, plus maybe a description of why it's useful to have a map vs. one value? This will help with answering the other questions Weilong posed below.

So I'm asking for comments like "lastUpdateTime maps endpointID to the timestamp of last create/update event for this endpoint"

Reply:
i don't understood is  that there contain two different conditions ,one for capturing CEP creat event, another for capturing CEP update event,  so why when i  add `oldEndpoint == nil && endpoint != nil` if contidion, then only capture creation, can't capture updation, could you explain more about it?

	if oldEndpoint == nil && endpoint != nil { 
		if p := k.endpointManager.LookupPodName(k8sUtils.GetObjNamespaceName(endpoint)); p != nil {
			timeSinceCepCreated := time.Since(p.GetCreatedAt())
			metrics.EndpointPropagationDelay.WithLabelValues().Observe(timeSinceCepCreated.Seconds())
		}
	} else if oldEndpoint != nil && endpoint != nil {

Reply:
hmm, somehow I did not see you added an else condition, maybe I was looking at an older version...but  now that there's an else capturing update, I'm okay with current code although  in my mind create and update should be simplified to a single timer, but that could a further optimization for the future :)